Makefile

```diff
@@ -182,13 +182,19 @@ docker-generate-k8s:
 		--env https_proxy=$(https_proxy) --env http_proxy=$(http_proxy) \
 		$(BUILD_IMAGE):$(OPERATOR_SDK_VERSION) /bin/bash -c 'operator-sdk generate k8s'
 
+SPEC_PROPS = .spec.versions[0].schema.openAPIV3Schema.properties.spec.properties
+
 docker-generate-crds:
 	echo "Generate crds"
 	docker run --rm -v $(PWD):$(WORKDIR) -v $(GOPATH)/pkg/mod:/go/pkg/mod:delegated \
 		-v $(shell go env GOCACHE):/root/.cache/go-build:delegated --env GO111MODULE=on \
 		--env https_proxy=$(https_proxy) --env http_proxy=$(http_proxy) \
 		$(BUILD_IMAGE):$(OPERATOR_SDK_VERSION) /bin/bash -c 'operator-sdk generate crds'
-	@sed -i '/\- protocol/d' deploy/crds/db.orange.com_cassandraclusters_crd.yaml
+	echo Update CRD - Remove protocol and set config type to object CRD
+	@sed -i -e '/\- protocol/d' deploy/crds/db.orange.com_cassandraclusters_crd.yaml
+	@yq -i e '$(SPEC_PROPS).config.type = "object"' deploy/crds/db.orange.com_cassandraclusters_crd.yaml
+	@yq -i e '$(SPEC_PROPS).topology.properties.dc.items.properties.config.type = "object"' deploy/crds/db.orange.com_cassandraclusters_crd.yaml
+	@yq -i e '$(SPEC_PROPS).topology.properties.dc.items.properties.rack.items.properties.config.type = "object"' deploy/crds/db.orange.com_cassandraclusters_crd.yaml
```

Comment:
This mean that there is not a strictly defined structure for the config, if I am correct.
I guess this is necessary since parameters can change over time / versions. But is it possible to maybe 'merge' CRD Yaml with cass-config-builder configs yamls (if they exist) to get all the different config parameters ?

Since this version cannot 'check' that a given config parameter is correct for the given Cassandra version, it will keep this (unfixable) issue BUT will prevent unknown configs (misstype for example).
What do you think ?

Reply:
There is no validation that can be done cause it depends on config-builder. I do it the same way Datastax does in their operator. there is no cass-config-builder CRD or so.